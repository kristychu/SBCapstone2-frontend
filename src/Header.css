.Header {
    background-color: #D291BC;
    color: #57175c;
    text-align: center;
    font-weight: bold;
  }
  
  .Header a {
    padding: 0.5em 0;
    width: 10em;
    display: inline-block;
    color: white;
    text-decoration: none;
  }
  
  .Header a:hover {
    background-color: #957DAD;
    color: white !important;
  }
  
  .Header a.active {
    font-weight: bold;
    color: #7f1d85;
  }

  .Header .logged-in {
    background-color: #f5bfe4;
  }