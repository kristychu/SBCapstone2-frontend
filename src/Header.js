import React from 'react';
import { NavLink, useHistory } from "react-router-dom";
import { useDispatch, useSelector } from 'react-redux';
import { logout } from './actions';
import './Header.css';

function Header(){
    const dispatch = useDispatch();
    const history = useHistory();
    
    const handleLogout = evt => {
        evt.preventDefault();
        dispatch(logout());
        history.push('/')
      };

    const token = useSelector(st => st.users.profile.token);
    if(token){
        return (
        <div className="Header">
            <h1>Korean Skincare App</h1>
            <nav>
                <NavLink exact to="/">
                    Home
                </NavLink>
                <NavLink exact to="/korean-skincare-routine">
                    The Routine
                </NavLink>
                <NavLink exact to="/user-routine" className="logged-in">
                    Your Routine
                </NavLink>
                <NavLink exact to="/logout" className="logged-in" onClick={handleLogout}>
                    Logout
                </NavLink>
            </nav>
        </div>
        )
    }
    return (
        <div className="Header">
            <h1>Korean Skincare App</h1>
            <nav>
                <NavLink exact to="/">
                    Home
                </NavLink>
                <NavLink exact to="/korean-skincare-routine">
                    The Routine
                </NavLink>
                <NavLink exact to="/register">
                    Register
                </NavLink>
                <NavLink exact to="/login">
                    Login
                </NavLink>
            </nav>
        </div>
      );
}

export default Header;