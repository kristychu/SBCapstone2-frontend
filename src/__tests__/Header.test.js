import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers/rootReducer";
import { render } from "@testing-library/react";
import Header from "../Header";
import { MemoryRouter } from "react-router";

const store = createStore(reducer);

it("should render without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </Provider>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
