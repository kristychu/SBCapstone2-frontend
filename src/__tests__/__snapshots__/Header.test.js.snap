// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for anonymous user 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <h1>
      Korean Skincare App
    </h1>
    <nav>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        href="/korean-skincare-routine"
      >
        The Routine
      </a>
      <a
        href="/register"
      >
        Register
      </a>
      <a
        href="/login"
      >
        Login
      </a>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot if logged in 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <h1>
      Korean Skincare App
    </h1>
    <nav>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        href="/korean-skincare-routine"
      >
        The Routine
      </a>
      <a
        class="logged-in"
        href="/user-routine"
      >
        Your Routine
      </a>
      <a
        class="logged-in"
        href="/logout"
      >
        Logout
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
